src/app/shared/mock-data.service.ts

```diff
@@ -186,6 +192,15 @@ export class MockDataService {
     return this.makeCopy(this.workItemLinks);
   }
 
+  public getWorkItemLinksForWorkItem(workItemId: string): any {
+    var result: any = [];
+    for (var i = 0; i < this.workItemLinks.length; i++)
+      if (this.workItemLinks[i].relationships.source.data.id === workItemId) {
+        result.push(this.makeCopy(this.workItemLinks[i]));
```

Comment:
In which cases? I would suspect if it fails, then the data structure is weird anyway, because the serialization must be done at some point anyway (on the rest request for example). So it would be interesting to see in which cases it fails.